### ISSUE TITLE
Timezone change in the future fails

### ISSUE THREAD

**Issue by tbee** [ical4j/ical4j#745]
If there is a VTIMEZONE with a zone change in the future, ZoneRulesBuilder line 86 will create a period with now as the end time:
```
observance.calculateRecurrenceSet(new Period<>(start.getDate(), LocalDateTime.now())).forEach( p -> {
```

Since some DTSTARTs in the zone are in the future, this fails in Interval:
```
throw new DateTimeException("End instant must be equal or after start instant");
````

Failing example (ical4j 4.0.4):
```java
class Scratch {
    public static void main(String[] args) throws ParseException, ParserException, IOException {
        String ical = """
                BEGIN:VCALENDAR
                VERSION:2.0
                CALSCALE:GREGORIAN
                METHOD:PUBLISH
                BEGIN:VTIMEZONE
                TZID:Europe/Amsterdam
                BEGIN:DAYLIGHT
                TZOFFSETFROM:+0100
                TZOFFSETTO:+0200
                TZNAME:CEST
                DTSTART:20240331T010000
                END:DAYLIGHT
                BEGIN:STANDARD
                TZOFFSETFROM:+0200
                TZOFFSETTO:+0100
                TZNAME:CET
                DTSTART:20241027T010000
                END:STANDARD
                BEGIN:DAYLIGHT
                TZOFFSETFROM:+0100
                TZOFFSETTO:+0200
                TZNAME:CEST
                DTSTART:20250330T010000
                END:DAYLIGHT
                BEGIN:STANDARD
                TZOFFSETFROM:+0200
                TZOFFSETTO:+0100
                TZNAME:CET
                DTSTART:20251026T010000
                END:STANDARD
                END:VTIMEZONE
                END:VCALENDAR                
                """;
        new CalendarBuilder().build(new StringReader(ical));
    }
}
```

Result:
```
15:05:59.654 [main] INFO  net.fortuna.ical4j.util.Configurator - ical4j.properties not found.
Exception in thread "main" net.fortuna.ical4j.data.ParserException: Error at line 31:End instant must be equal or after start instant
	at net.fortuna.ical4j.data.CalendarParserImpl.parse(CalendarParserImpl.java:167)
	at net.fortuna.ical4j.data.CalendarBuilder.build(CalendarBuilder.java:197)
	at net.fortuna.ical4j.data.CalendarBuilder.build(CalendarBuilder.java:185)
	at Scratch.main(scratch_1.java:49)
Caused by: java.time.DateTimeException: End instant must be equal or after start instant
	at org.threeten.extra.Interval.of(Interval.java:105)
	at net.fortuna.ical4j.model.Period.toInterval(Period.java:486)
	at net.fortuna.ical4j.model.Period.toInterval(Period.java:470)
	at net.fortuna.ical4j.model.Period.intersects(Period.java:464)
	at net.fortuna.ical4j.model.Component.calculateRecurrenceSet(Component.java:373)
	at net.fortuna.ical4j.model.ZoneRulesBuilder.buildDSTTransitions(ZoneRulesBuilder.java:86)
	at net.fortuna.ical4j.model.ZoneRulesBuilder.build(ZoneRulesBuilder.java:145)
	at net.fortuna.ical4j.model.TimeZoneRegistryImpl.register(TimeZoneRegistryImpl.java:158)
	at net.fortuna.ical4j.model.TimeZoneRegistryImpl.register(TimeZoneRegistryImpl.java:136)
	at net.fortuna.ical4j.data.DefaultContentHandler.endComponent(DefaultContentHandler.java:125)
	at net.fortuna.ical4j.data.CalendarParserImpl$ComponentParser.parse(CalendarParserImpl.java:449)
	at net.fortuna.ical4j.data.CalendarParserImpl$PropertyListParser.parse(CalendarParserImpl.java:217)
	at net.fortuna.ical4j.data.CalendarParserImpl.parseCalendar(CalendarParserImpl.java:126)
	at net.fortuna.ical4j.data.CalendarParserImpl.parseCalendarList(CalendarParserImpl.java:188)
	at net.fortuna.ical4j.data.CalendarParserImpl.parse(CalendarParserImpl.java:161)
	... 3 more
```

**Comment by tbee** [comment]
I was making a fix and saw that on the development branch this is already fixed. When will this be released?

**Comment by tbee** [comment]
Hm. Using the version on the development branch, I run into the issue that the DateProperty has a format of yyyyMMdd'T'HHmmss'Z', but there is a DTSTAMP being process that has the value of 'now' without the Z (20241004T153715). Need to figure out where that comes from, but the my problem has moved.

```
java.time.format.DateTimeParseException: Text '20241004T152911' could not be parsed at index 15
	at java.base/java.time.format.DateTimeFormatter.parseResolved0(DateTimeFormatter.java:2052)
	at java.base/java.time.format.DateTimeFormatter.parse(DateTimeFormatter.java:1954)
	at net.fortuna.ical4j.model.CalendarDateFormat.parse(CalendarDateFormat.java:293)
	at net.fortuna.ical4j.model.TemporalAdapter.parse(TemporalAdapter.java:250)
	at net.fortuna.ical4j.model.property.DateProperty.setValue(DateProperty.java:200)
	at net.fortuna.ical4j.model.property.DtStamp.<init>(DtStamp.java:110)
	at net.fortuna.ical4j.model.property.DtStamp$Factory.createProperty(DtStamp.java:162)
	at net.fortuna.ical4j.model.property.DtStamp$Factory.createProperty(DtStamp.java:153)
	at net.fortuna.ical4j.model.PropertyBuilder.build(PropertyBuilder.java:110)
	at net.fortuna.ical4j.data.DefaultContentHandler.endProperty(DefaultContentHandler.java:153)
	at net.fortuna.ical4j.data.CalendarParserImpl$PropertyParser.parse(CalendarParserImpl.java:293)
	at net.fortuna.ical4j.data.CalendarParserImpl$PropertyListParser.parse(CalendarParserImpl.java:219)
	at net.fortuna.ical4j.data.CalendarParserImpl$ComponentParser.parse(CalendarParserImpl.java:427)
	at net.fortuna.ical4j.data.CalendarParserImpl$PropertyListParser.parse(CalendarParserImpl.java:217)
	at net.fortuna.ical4j.data.CalendarParserImpl.parseCalendar(CalendarParserImpl.java:126)
	at net.fortuna.ical4j.data.CalendarParserImpl.parseCalendarList(CalendarParserImpl.java:188)
	at net.fortuna.ical4j.data.CalendarParserImpl.parse(CalendarParserImpl.java:161)
	at net.fortuna.ical4j.data.CalendarBuilder.build(CalendarBuilder.java:197)
	at net.fortuna.ical4j.data.CalendarBuilder.build(CalendarBuilder.java:185)
```

**Comment by tbee** [comment]
So the DTSTAMP containing 'now' is apparently generated by the server providing the ics file. And it never has the Z suffix. This used to be parsed correctly by ical4j 3.

**Comment by benfortuna** [comment]
I think we can support using RELAXED_DATE_FORMAT for parsing DTSTAMP when the RELAXED_PARSING compatibility hint is enabled. I'll include the change in the next release.

**Comment by tbee** [comment]
That would be nice. For now I've patched the incoming stream with some regexp, but I do need the changes that are currently in master. When will a new release be made?

**Comment by benfortuna** [comment]
>  When will a new release be made?

Already released, in 4.0.5 today
